src/main/java/model/UMLClassManager.java

```diff
@@ -413,34 +466,47 @@ public int removeRelationship(String srcClass, String type, String destClass) {
 		return ErrorHandler.setCode(0);
 	}
 	
+	/**
+	 * Lists relationships of the entire model
+	 */
+	public ArrayList<ArrayList<String[]>> printRelationships() {
+		if(relationships.isEmpty()) {
+			return null;
+		}
+		else {
+			ArrayList<ArrayList<String[]>> result = new ArrayList<ArrayList<String[]>>();
+			for(Map.Entry<String, UMLRelationship> entry : relationships.entrySet()) {

```

Comment:
This is likely what is causing your double print issue. I don't see why there would need to be a nested for loop here, unless I'm misreading this.